### PR TITLE
Cleanup Library presenter

### PR DESCRIPTION
**Pull request by AntsyLich**
- Remove usage of `rx.Observable`
- Remove redundant [requestSortUpdate]
- Move [getTracksFlow] to domain
- Cleanup [getCommonCategories], [getMixCategories], [setMangaCategories]

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt [hunk 1]
```diff
@@ -110,15 +105,8 @@ class LibraryPresenter(
...
-    /**
-     * Relay used to apply the UI filters to the last emission of the library.
-     */
-    private val filterTriggerRelay = BehaviorRelay.create(Unit)
-
-    /**
-     * Relay used to apply the selected sorting method to the last emission of the library.
-     */
-    private val sortTriggerRelay = BehaviorRelay.create(Unit)
+    private val _filterChanges: MutableStateFlow<Int> = MutableStateFlow(0)
```
Comment (ghostbear) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt:108]: Use a Channel :harold:

Comment (AntsyLich) [reply app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt:108]: Doesn't seem to work if i keep sending the same value.

Comment (ghostbear) [reply app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryPresenter.kt:108]: Works fine if you keep sending the Unit object

**Review by ghostbear — APPROVED** [review]

File: app/src/main/java/eu/kanade/domain/track/interactor/GetLoggedTracksPerManga.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    private val trackManager: TrackManager,
+) {
+
+    fun subscribe(): Flow<Map<Long, Map<Long, Boolean>>> {
+        return trackRepository.getTracksAsFlow().map { tracks ->
+            tracks
+                .groupBy { it.mangaId }
+                .mapValues { entry ->
+                    entry.value.associate { track ->
+                        val isLogged = trackManager.getService(track.syncId)?.isLogged == true
```
Comment (arkon) [inline app/src/main/java/eu/kanade/domain/track/interactor/GetLoggedTracksPerManga.kt:19]: Do we even care about the items for trackers that aren't logged in?

**Pull request merged**